### PR TITLE
Add basic version of data collection api with CICD pipeline

### PR DESCRIPTION
**Pull request by anmolbabu**

### CODE REVIEW

**Comment by anmolbabu** [comment]
TODOs: 
1. All references to docker hub registry in pipeline/jenkins files need to changed to registry.devshift.net 
    along with names of jenkins ENV variables that store credentials..
    Note: They are currently hub.docker.com for ease of verification 
2. Currently the UTs require the supervisor app to be running but I think its better to deploy a mock 
    server from the main/base UT and all UTs use that...
3. Deployment aspects in the form of deployment resource templates and jenkfile entries corresponding 
    to deployment

Mitch(Sorry I couldn't find github name), @miteshvp , @mik-dass , @dev-gaur This is an intial WIP commit and currently in its early nascent stages Please review and provide your comments/feedback...

File: Dockerfile
Comment (miteshvp) [inline Dockerfile]: You can be specific with both the files.

File: Dockerfile
Comment (miteshvp) [inline Dockerfile]: Can also add a maintainer

**Review by miteshvp — APPROVED** [review]
Just couple of minor nitpick, otherwise LGTM. Nice work.

File: Dockerfile
Comment (anmolbabu) [reply Dockerfile]: Ack

File: Dockerfile
Comment (anmolbabu) [reply Dockerfile]: You mean like:
COPY package*.json ./package*.json
right?
Ok

Comment (miteshvp) [reply Dockerfile]: Yes

Comment (anmolbabu) [reply Dockerfile]: Ack
Done

**Comment by miteshvp** [comment]
@anmolbabu - Just one more thing. If its a WIP, I suggest changing the title of this PR by preponing [WIP] to it.

**Comment by anmolbabu** [comment]
@miteshvp ack
Done

**Comment by miteshvp** [comment]
@anmolbabu - if you're done, please remove WIP title and I will merge this.

**Comment by anmolbabu** [comment]
@miteshvp Ack Done..
As discussed I'll raise the issues for ToDos above for tracking purpose.

**Comment by anmolbabu** [comment]
Created the following issues for better tracking to avoid blocking this PR(to unblock Peers from sending PRs):
1. https://github.com/fabric8-hdd/supervisor/issues/2
2. https://github.com/fabric8-hdd/supervisor/issues/3
3. https://github.com/fabric8-hdd/supervisor/issues/4

**Comment by miteshvp** [comment]
LGTM. Merged

**Pull request merged**
